manifest.yaml

```diff
@@ -172,6 +172,21 @@ interfaces:
     protocols:
       - tcp
       - http
+  websocket:
+    name: Clams Websocket
+    description: Websocket endpoint for Clams Remote.
+    tor-config:
+      port-mapping:
+        7272: "7272"
+    lan-config:
+      7272:
+        ssl: true
+        internal: 7272
+    ui: false
+    protocols:
+      - tcp
+      - http
+      - ws
```

Comment:
So there are two onion endpoints that will be exposed in this project. This this particular repo, we expose Core Lightning's websocket as an onion endpoint. This allows any Clams Remote client running in Tor Browser to directly connect back to and control Core Lightning via websocket.

The second onion endpoint is the web server that serves Clams Remote static build output. This is addressed in another repository that I'll be making public soon.

Reply:
Connecting Clams to another service (CLN) on the same server using the tor network seems inefficient and undesirable. Unless there is a Clams limitation I am not understanding that would prevent this, I would think it would be far more desirable for Clams to connect locally to the Core Lightning container, `c-lightning.embassy`.

Looking at the Clams [documentation for testing locally](https://docs.clams.tech/remote/#testing-locally), I believe you should be able to get Clams connect to `https://c-lightning.embassy:7272` assuming port 7272 has been exposed.

Reply:
Ok I think I see what you're saying now. As far as I know, that is not the model of Clams Remote. remote.clams.tech is just a web server that for the most part, serves static HTML/CSS/JS to requesting parties (Clams Remote a PWA). The container running the web server that serves these files should NOT require access to any RPC interfaces on the c-lightning container (e.g., via network bridge), nor should it require access to any .lightning volumes. It's purpose is to simply serve the Clams Remote PWA to request web browser clients.

Once the PWA is loaded in the end-users browser, users can connect to one or more CLN nodes. The ONLY way to achieve this in Sovereign manner is to connect directly to your node. The method Clams Remote supports for this is direct websocket. This is configured in the users browser context, along with any authorization/rune information. This preferable from a security point of view that we know that connection/authorization information to your node is held exclusively on the user's device.

Reply:
Btw I pushed some commits to my implementation to the Clams Remote. https://github.com/farscapian/clams-startos . That works in conjunction with this pull request as described.

Reply:
To confirm, you have tested the package of Clams Remote with this PR, and it is working as expected?

Connecting Clams to CLN over tor when they are running on the same box is not ideal, but this seems to be a limitation of Clams at present.

Reply:
Yes, the `clams-startos` repo I posted above works in conjunction with this pull request and works as expected.